ios/FluentUI.Demo/FluentUI.Demo/Demos/AvatarGroupDemoController.swift

```diff
@@ -0,0 +1,289 @@
+        var constraints: [NSLayoutConstraint] = []
+
+        for size in MSFAvatarSize.allCases.reversed() {
+            let containerView = UIView(frame: .zero)
+
+            let avatarGroup = MSFAvatarGroup(style: style, size: size)
+            var border = showBorders
+            for index in 0...avatarCount - 1 {
+                vSamplePersonas[index].isRingVisible = border
+                if mixed {
+                    border = !border
+                }
+
+                let avatarState = avatarGroup.state.createAvatar(style: .default, size: size)
+                let samplePersona = vSamplePersonas[index]
+
+                avatarState.accessibilityLabel = samplePersona.accessibilityLabel
+                avatarState.backgroundColor = samplePersona.backgroundColor
+                avatarState.foregroundColor = samplePersona.foregroundColor
+                avatarState.hasPointerInteraction = samplePersona.hasPointerInteraction
+                avatarState.hasRingInnerGap = samplePersona.hasRingInnerGap
+                avatarState.image = samplePersona.image
+                avatarState.imageBasedRingColor = samplePersona.imageBasedRingColor
+                avatarState.isOutOfOffice = samplePersona.isOutOfOffice
+                avatarState.isRingVisible = border
+                avatarState.isTransparent = samplePersona.isTransparent
+                avatarState.presence = samplePersona.presence
+                avatarState.primaryText = samplePersona.primaryText
+                avatarState.ringColor = samplePersona.ringColor
+                avatarState.secondaryText = samplePersona.secondaryText
+            }
+
+            avatarGroup.state.maxDisplayedAvatars = maxDisplayedAvatars
+            avatarGroup.state.overflowCount = overflowCount
+            avatarGroups.append(avatarGroup)
+            let avatarGroupView = avatarGroup.view
+            avatarGroupView.translatesAutoresizingMaskIntoConstraints = false
+            containerView.addSubview(avatarGroupView)
+
+            let trailingConstraint = containerView.trailingAnchor.constraint(equalTo: container.trailingAnchor)
+            trailingConstraint.priority = .defaultHigh
+
+            constraints.append(contentsOf: [
+                avatarGroupView.leadingAnchor.constraint(equalTo: containerView.leadingAnchor),
+                avatarGroupView.topAnchor.constraint(equalTo: containerView.topAnchor),
+                avatarGroupView.bottomAnchor.constraint(equalTo: containerView.bottomAnchor),
+                trailingConstraint
+            ])
+
+            addRow(text: size.description, items: [containerView], textStyle: .footnote, textWidth: 100)
+        }
+
+        NSLayoutConstraint.activate(constraints)
+    }
+
+    private var vSamplePersonas: [MSFAvatarState] = {
+        var personas: [MSFAvatarState] = []
+        for index in 0...samplePersonas.count - 1 {
+            let state = MSFAvatar().state
```

Comment:
This is not how the state should be used or created.

The samplePersonas (which is just the data class) should be iterated and createAvatar should be called on each avatarGroup state to create the avatars. Creating avatarStates here is just duplicating the data storage for no added benefit.